### PR TITLE
CCTextFieldTTF: secureTextEntry property

### PR DESCRIPTION
**Pull request by MarcelBloemendaal**
Added a secureTextEntry property to CCTextFieldTTF. Enabling this property causes the
TextField to display the entered text as bullets.

(I'm not sure those issues are still in use, but it used to be issue #730 on the cocos2d-x site)

### CODE REVIEW

File: cocos2dx/text_input_node/CCTextFieldTTF.h [hunk 1]
```diff
@@ -145,6 +145,11 @@ class CC_DLL CCTextFieldTTF : public CCLabelTTF, public CCIMEDelegate
 protected:
     std::string * m_pPlaceHolder;
     ccColor3B m_ColorSpaceHolder;
+public:
+	virtual void setSecureTextEntry(bool value);
+	virtual bool getSecureTextEntry();
+protected:
+	bool m_pSecureTextEntry;
```
Comment (dumganhar) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: Why to name this variable `m_pSecureTextEntry`?
`m_bSecureTextEntry` will be more suitable.
Also, the method `getSecureTextEntry()` should be `isSecureTextEntry()`.

Comment (MarcelBloemendaal) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: Hello James,

I'm not sure what the naming conventions are for cocos. I was assuming the p stands for 'property'. Why should it be m_bSecureTextEntry? (I'd like to understand, so next time I can use proper conventions)

I do agree on your second point: isSecureTextEntry is probably better. Should I change this and create a new pull request? Or will you?

Grtz,

Marcel Bloemendaal.

Comment (dumganhar) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: > I'm not sure what the naming conventions are for cocos. I was assuming the p stands for 'property'. Why should it be m_bSecureTextEntry? (I'd like to understand, so next time I can use proper conventions)

The `b` means it's a boolean value. Let me find out the naming conventions for cocos2dx later. :)

> Should I change this and create a new pull request?

You just need to push you changes to your original remote repo . This pull request will be auto updated. Thanks.

**Comment by dumganhar** [comment]
Please refer to http://www.learncpp.com/cpp-tutorial/29-hungarian-notation/
and http://www.cocos2d-x.org/projects/cocos2d-x/wiki/Cocos_Coding_Style
Thank you.

File: cocos2dx/text_input_node/CCTextFieldTTF.h (hunk 1 above)
Comment (dumganhar) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: When will this Pull  Request be updated?

Comment (MarcelBloemendaal) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: Oops, must have forgotten to push it.. will do so tomorrow at work!

Sorry!

Comment (dumganhar) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: OK, thanks.

Comment (MarcelBloemendaal) [inline cocos2dx/text_input_node/CCTextFieldTTF.h]: Should be done now!

Op 14 apr. 2013 om 11:32 heeft James Chen notifications@github.com het
volgende geschreven:

In cocos2dx/text_input_node/CCTextFieldTTF.h:

> @@ -145,6 +145,11 @@ class CC_DLL CCTextFieldTTF : public CCLabelTTF, public CCIMEDelegate
>  protected:
>      std::string \* m_pPlaceHolder;
>      ccColor3B m_ColorSpaceHolder;
> +public:
> - virtual void setSecureTextEntry(bool value);
> - virtual bool getSecureTextEntry();
>   +protected:
> - bool m_pSecureTextEntry;

OK, thanks.

—
Reply to this email directly or view it on
GitHubhttps://github.com/cocos2d/cocos2d-x/pull/2277/files#r3785539
.

File: cocos2dx/text_input_node/CCTextFieldTTF.cpp [hunk 2]
```diff
@@ -287,11 +288,25 @@ void CCTextFieldTTF::setColorSpaceHolder(const ccColor3B& color)
...
     {
         m_pInputText = new std::string(text);
+		displayText = *m_pInputText;
+		if (m_bSecureTextEntry)
+		{
+			displayText = "";
+			length = m_pInputText->length();
+			while (length)
+			{
+				displayText.append("•");
```
Comment (dumganhar) [inline cocos2dx/text_input_node/CCTextFieldTTF.cpp:306]: `"•"` is a utf8 string, if you compile this file on windows by Visual Studio, a compilation error will be triggered.

Comment (MarcelBloemendaal) [inline cocos2dx/text_input_node/CCTextFieldTTF.cpp:306]: Ouch.. Ok, that would be a problem. I'm afraid I don't really know much
about the windows platform. What would you suggest? We could substute for a
- character, but it's less pretty...

Op 18 apr. 2013 om 12:39 heeft James Chen notifications@github.com het
volgende geschreven:

In cocos2dx/text_input_node/CCTextFieldTTF.cpp:

> ```
>  CC_SAFE_DELETE(m_pInputText);
> 
>  if (text)
>  {
>      m_pInputText = new std::string(text);
> ```
> -     displayText = *m_pInputText;
> -     if (m_bSecureTextEntry)
> -     {
> -         displayText = "";
> -         length = m_pInputText->length();
> -         while (length)
> -         {
> -             displayText.append("•");

"•" is a utf8 string, if you compile this file on windows by Visual Studio,
a compilation error will be triggered.

—
Reply to this email directly or view it on
GitHubhttps://github.com/cocos2d/cocos2d-x/pull/2277/files#r3850862
.

**Comment by dumganhar** [comment]
Merged at https://github.com/cocos2d/cocos2d-x/pull/2403.
Thanks.

**Pull request merged**